doc/source/data/working-with-tensors.rst

```diff
@@ -69,6 +69,42 @@ Elements of ragged arrays are regular ndarrays.
     >>> batch["image"][3].shape  # doctest: +SKIP
     (333, 465, 3)
 
+.. _transforming_tensors:
+
+Transforming tensor data
+------------------------
+
+Call :meth:`~ray.data.Dataset.map` or :meth:`~ray.data.Dataset.map_batches` to transform tensor data.
+
+.. testcode::
+
+    from typing import Any, Dict
+
+    import ray
+    import numpy as np
+
+    ds = ray.data.read_images("s3://anonymous@air-example-data/AnimalDetection")
+
+    def increase_brightness(row: Dict[str, Any]) -> Dict[str, Any]:
+        row["image"] = np.clip(row["image"] + 4, 0, 255)
+        return row
+
+    # Increase the brightness, record at a time.
+    ds.map(increase_brightness)
+
+    def batch_increase_brightness(batch: Dict[str, np.ndarray]) -> Dict:
+        batch["image"] = np.clip(batch["image"] + 4, 0, 255)
+        return batch
+
+    # Increase the brightness, batch at a time.
+    ds.map_batches(batch_increase_brightness)
+
+In this example, we return ``np.ndarray`` directly as the output. Ray Data will also treat
+returned lists of ``np.ndarray`` and objects implementing ``__array__`` as tensor data.
```

Comment:
add some examples of object implementing the array protocol (pytorch tensors, etc.)?

I don't think many people know what this actually means